### PR TITLE
community : [bugfix] Use document ids as keys in AzureSearch vectorstore

### PR DESCRIPTION
**Pull request by MacanPN**
# Description
[Vector store base class](https://github.com/langchain-ai/langchain/blob/4cdaca67dc51dba887289f56c6fead3c1a52f97d/libs/core/langchain_core/vectorstores/base.py#L65) currently expects `ids` to be passed in and that is what it passes along to the AzureSearch vector store when attempting to `add_texts()`. However AzureSearch expects `keys` to be passed in. When they are not present, AzureSearch `add_embeddings()` makes up new uuids. This is a problem when trying to run indexing. [Indexing code expects](https://github.com/langchain-ai/langchain/blob/b297af5482ae7c6d26779513d637ec657a1cd552/libs/core/langchain_core/indexing/api.py#L371) the documents to be uploaded using provided ids. Currently AzureSearch ignores `ids` passed from `indexing` and makes up new ones. Later when `indexer` attempts to delete removed file, it uses the `id` it had stored when uploading the document, however it was uploaded under different `id`.

**Twitter handle: @martintriska1**

### CODE REVIEW

**Comment by MacanPN** [comment]
@eyurtsev please review or assign to someone! Thanks!

**Comment by MacanPN** [comment]
Is there anything I can do to move this PR forward? Please let me know!

**Comment by MacanPN** [comment]
Pretty please, could someone please take a look? We're using this fix in our pipeline and have to keep using forked version until the PR gets through. Please let me know if I can do anything to speed up the process!

@baskaryan @efriis  @eyurtsev @ccurme @vbarda @hwchase17

**Review by ccurme — COMMENTED** [review]
The change looks fine to me, but could you add a test to the existing `AzureSearch` tests that fails with existing code, and passes following this update?

**Comment by MacanPN** [comment]
@ccurme added a unit test. Please take a look. Thanks!

File: libs/community/langchain_community/vectorstores/azuresearch.py [hunk 1]
```diff
@@ -468,9 +480,13 @@ def add_embeddings(
...
             # Use provided key otherwise use default key
-            key = keys[i] if keys else str(uuid.uuid4())
-            # Encoding key for Azure Search valid characters
-            key = base64.urlsafe_b64encode(bytes(key, "utf-8")).decode("ascii")
+            if keys:
+                key = keys[i]
+            else:
+                key = str(uuid.uuid4())
+                # Encoding key for Azure Search valid characters
+                key = base64.urlsafe_b64encode(bytes(key, "utf-8")).decode("ascii")
```
[the remark this answers is not in the document]
Comment (MacanPN) [reply libs/community/langchain_community/vectorstores/azuresearch.py:488]: @ccurme I'm sorry but I replied to your question 4 days ago but didn't hit "start review" (I assumed the comment would be still visible). Only today I found out that the comment does not show up if the review is not "started".
Please take a look and let me know whether it answers your questions and whether you agree. Thanks!

[the remark this answers is not in the document]
Comment (MacanPN) [reply libs/community/langchain_community/vectorstores/azuresearch.py:488]: @ccurme Please let me know if I can expand on the explanation I gave. Or please let me know what can I do to move forward. The issue I'm trying to fix is that as it stands right now, the indexer does not work with Azure at all. Thank you!

**Comment by petergoldstein** [comment]
@ccurme Are you available to give this another look?  I think @MacanPN 's responses address your question. 

This is a blocker for me, and I'd really like to see it merged.  If there's anything I can do to move that along, please let me know.  Thanks!

**Review by ccurme — APPROVED** [review]

**Pull request merged**